research-software-directory/RSD-as-a-service
# feat: edit contributor and team member modal options

# Edit contributor and team member

Closes #450
Closes #1152 

Changes proposed in this pull request:
* edit contributor avatar, roles and affiliation using aggregated values from other RSD entries. 
* edit team member avatar, roles and affiliation using aggregated values from other RSD entries.
* Avatar and affiliation values are aggregated based on contributor/team member ORCID.
* use of "person" shared components for "edit" modal and search/find option
* moved methods to "hook" files
* FIX: removed **fuzzy** search "indicator" from ORCID search url (because it does not produce expected behaviour).
* FIX: **improve deduplication at import of contributors**, use last_name+first name instead of only last_name.
* rsd admin can update rsd contributor avatars. The avatars are aggregated on "display_name" (first name + last name) or ORCID.

How to test:
* `make start` to build and generate test data
* login as rsd admin
* navigate to edit software page, contributors section.
* test all options: create, find, edit and delete contributor. You should be able to select avatar images, roles and affiliation based on the data already present in RSD
* test contributors with ORCID and without ORCID
* navigate to edit project, team members section
* test all options: create, find, edit and delete contributor. You should be able to select avatar images, roles and affiliation based on the data already present in RSD
* test team members with ORCID and without ORCID
* test rsd admin edit avatars section:
   *  first add yourself to number of software and project items and upload different images
   * then navigate to rsd admin, rsd contributors page
   * search for yourself by first or last name. You should see a list of your entries. In the avatar column you will see avatar and the count of options. If there is a more than 1 avatar option you will be able to click on avatar and change it.

## Contributor modal (example)
![image](https://github.com/user-attachments/assets/f7003724-167b-42f6-96d2-44c5cc176132)

## Team member modal (example)
![image](https://github.com/user-attachments/assets/5a2e6ed7-7b6e-41ba-9a8c-becc01253c55)

## Rsd admin - contributors
![image](https://github.com/user-attachments/assets/beb524a9-cf34-4d2e-87e1-665f86e973b2)


PR Checklist:

*
